### PR TITLE
Update sample app to last Core API version

### PR DESCRIPTION
**Pull request by jcharnay38**
- Related to: https://github.com/nabla/health/issues/37479

---

- Applied the detailed migration guide, to migrate from version prior to 2024-10-01 to last 2024-11-08 API version: https://docs.nabla.com/guides/api-versioning/migrating-from-prior-to-2024-10-01
- Used new more secure OAuth authentication flow instead of server API keys: https://docs.nabla.com/guides/authentication
  - to construct the JWT Client Assertion, I needed the **jsrassign** library (cryptographic signing cannot be done easily in vanilla JS). To keep the sample app simple and standalone, I decided to integrate the library code directly into this repo, in the form of a minified standalone JS file.
- Updated README accordingly

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -25,38 +26,32 @@ <h1>Nabla API sample app</h1>
...
                 <label for="transcript-locale">Transcript locale:</label>
                 <select id="transcript-locale">
-                    <option value="en-US">en-US</option>
-                    <option value="en-GB">en-GB</option>
-                    <option value="fr-FR">fr-FR</option>
-                    <option value="es-ES">es-ES</option>
-                    <option value="es-MX">es-MX</option>
+                    <option value="ENGLISH_US">ENGLISH_US</option>
+                    <option value="ENGLISH_UK">ENGLISH_UK</option>
+                    <option value="FRENCH">FRENCH</option>
```
Comment (AchrafAmil) [inline index.html]: It's `FRENCH_FR`, also for transcript-locale we can now list all the 35 supported locales.
```suggestion
                    <option value="FRENCH_FR">FRENCH_FR</option>
```

File: index.html [hunk 2]
```diff
@@ -25,38 +26,32 @@ <h1>Nabla API sample app</h1>
...
 
             <div class="filters">
                 <label for="note-locale">Note locale:</label>
                 <select id="note-locale">
-                    <option value="en-US">en-US</option>
-                    <option value="en-GB">en-GB</option>
-                    <option value="fr-FR">fr-FR</option>
+                    <option value="ENGLISH_US">ENGLISH_US</option>
+                    <option value="ENGLISH_UK">ENGLISH_UK</option>
+                    <option value="FRENCH">FRENCH</option>
```
Comment (AchrafAmil) [inline index.html]:
```suggestion
                    <option value="FRENCH_FR">FRENCH_FR</option>
```

File: main.js [hunk 3]
```diff
@@ -9,6 +11,57 @@ let mediaStream
...
+    if (
+        serverAccessTokenCache.accessToken &&
+        nowSeconds < serverAccessTokenCache.expiresAt
+    ) {
+        return serverAccessTokenCache.accessToken;
+    }
+
+    const { access_token, expires_in } = await fetchServerAccessToken();
+    serverAccessTokenCache.accessToken = access_token;
+    serverAccessTokenCache.expiresAt = nowSeconds + expires_in - 5;
```
Comment (AchrafAmil) [inline main.js]: nit but I'd move the `- 5` to the moment where you read the value (as a leeway for that specific usage) and keep the stored `expiresAt` exactly equal to `nowSeconds + expires_in`

**Review by AchrafAmil — APPROVED** [review]
Awesome 🚀

**Review by gregoirenovel — COMMENTED** [review]
OK product

**Pull request merged**